pkg/forwarder/failed_transaction_removal_policy.go

```diff
@@ -0,0 +1,141 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package forwarder
+
+import (
+	"crypto/md5"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/util"
+	"github.com/hashicorp/go-multierror"
+)
+
+type failedTransactionRemovalPolicy struct {
+	rootPath           string
+	knownDomainFolders map[string]struct{}
+	outdatedFileTime   time.Time
+}
+
+func newFailedTransactionRemovalPolicy(rootPath string, outdatedFileDayCount int) (*failedTransactionRemovalPolicy, error) {
+	if err := os.MkdirAll(rootPath, 0755); err != nil {
+		return nil, err
+	}
+
+	return &failedTransactionRemovalPolicy{
+		rootPath:           rootPath,
+		knownDomainFolders: make(map[string]struct{}),
+		outdatedFileTime:   time.Now().Add(time.Duration(-outdatedFileDayCount*24) * time.Hour),
+	}, nil
+}
+
+// RegisterDomain registers a domain name.
+func (p *failedTransactionRemovalPolicy) RegisterDomain(domainName string) (string, error) {
```

Comment:
This should be private since it's not use outside the forwarder and it's on a private struct.

Reply:
Code updated